include/arch/x86/page.h

```diff
@@ -287,15 +290,15 @@ static inline void *kmap_4k(mfn_t mfn, unsigned long l1_flags) {
 static inline void *vmap_user_1g(void *va, mfn_t mfn, unsigned long l3_flags) {
     unsigned long user = l3_flags & _PAGE_USER;
 
-    return vmap_user(va, mfn, PAGE_ORDER_1G, L4_PROT | user, l3_flags | user, PT_NO_FLAGS,
-                     PT_NO_FLAGS);
+    return vmap_user(va, mfn, PAGE_ORDER_1G, L4_PROT | user,
+                     l3_flags | (user | _PAGE_PSE), PT_NO_FLAGS, PT_NO_FLAGS);
 }
 
 static inline void *vmap_user_2m(void *va, mfn_t mfn, unsigned long l2_flags) {
     unsigned long user = l2_flags & _PAGE_USER;
 
     return vmap_user(va, mfn, PAGE_ORDER_2M, L4_PROT | user, L3_PROT | user,
-                     l2_flags | user, PT_NO_FLAGS);
+                     l2_flags | (user | _PAGE_PSE), PT_NO_FLAGS);
```

Comment:
parentheses again, hm are they intentional?

Reply:
Yes, I was aiming for separating multiple newly applied flags for readability.